src/scalar.rs

```diff
@@ -989,29 +989,30 @@ impl Scalar {
     /// Returns a size hint indicating how many entries of the return
     /// value of `to_radix_2w` are nonzero.
     pub(crate) fn to_radix_2w_size_hint(w: usize) -> usize {
-        debug_assert!(w >= 6);
+        debug_assert!(w == 4 || w >= 6);
         debug_assert!(w <= 8);
 
         let digits_count = match w {
+            4 => (256 + w - 1)/w as usize,
+            5 => (256 + w - 1)/w as usize,
```

Comment:
Is the w = 5 case supposed to be reached and handled properly? (If not, then why would this case fail in the debug mode for `debug_assert!(w == 4 || w >= 6);` ?

It seems like undesirable behavior for the w=5 case to fail in the debug mode but be handled correctly in the optimized mode, but maybe I am missing something?

Reply:
Oh nope, good catch. There was an older version of my code which only handled 4, 6, 7, and 8, and I fixed the debug assert in the function beneath but not this one. Thanks!

Reply:
Fixed in ca1f730

Reply:
👍